packages/alfa-dom/src/tables/cell.ts

```diff
@@ -0,0 +1,730 @@
+    element: Element,
+    headers: Array<Element> = []
+  ): Cell {
+    return new Cell(kind, x, y, width, height, element, headers);
+  }
+
+  private constructor(
+    kind: Cell.Kind,
+    x: number,
+    y: number,
+    width: number,
+    height: number,
+    element: Element,
+    headers: Array<Element>
+  ) {
+    this._kind = kind;
+    this._x = x;
+    this._y = y;
+    this._width = width;
+    this._height = height;
+    this._element = element;
+    this._headers = headers;
+  }
+
+  // debug
+  public get name(): string {
+    return this._element.attribute("id").get().value;
+  }
+
+  public get anchor(): { x: number; y: number } {
+    return { x: this._x, y: this._y };
+  }
+
+  public get width(): number {
+    return this._width;
+  }
+
+  public get height(): number {
+    return this._height;
+  }
+
+  public get kind(): Cell.Kind {
+    return this._kind;
+  }
+
+  public get element(): Element {
+    return this._element;
+  }
+
+  public get headers(): Iterable<Element> {
+    return this._headers;
+  }
+
+  public isCovering(x: number, y: number): boolean {
+    return !(
+      // cell is *not* covering if either
+      (
+        x < this._x || // slot is left of cell
+        this._x + this._width - 1 < x || // slot is right of cell
+        y < this._y || // slot is above cell
+        this._y + this._height - 1 < y
+      ) // slot is below cell
+    );
+  }
+
+  /**
+   * compare cell according to their anchor
+   * in a given group of cells (row, rowgroup, table, …), no two different cells can have the same anchor, so this is good.
+   */
+  public compare(cell: Cell): Comparison {
+    if (this._y < cell._y) return Comparison.Smaller;
+    if (this._y > cell._y) return Comparison.Greater;
+    if (this._x < cell._x) return Comparison.Smaller;
+    if (this._x > cell._x) return Comparison.Greater;
+    return Comparison.Equal;
+  }
+
+  public equals(value: unknown): value is this {
+    return (
+      value instanceof Cell &&
+      this._kind === value._kind &&
+      this._width === value._width &&
+      this._height === value._height &&
+      this._x === value._x &&
+      this._y === value._y &&
+      this._element.equals(value._element)
+    );
+  }
+
+  public toJSON(): Cell.JSON {
+    return {
+      kind: this._kind,
+      anchor: this.anchor,
+      width: this._width,
+      height: this._height,
+      element: this.name, // this._element.toJSON(),
+      headers: this._headers.map(
+        (header) => header.attribute("id").get().value
+      ), // this._headers.map((header) => header.toJSON()),
+    };
+  }
+}
+
+export namespace Cell {
+  export interface JSON {
+    [key: string]: json.JSON;
+
+    kind: Kind;
+    anchor: { x: number; y: number };
+    width: number;
+    height: number;
+    element: string; //Element.JSON;
+    headers: string[]; //Element.JSON[];
+  }
+
+  export enum Kind {
+    Header = "header",
+    Data = "data",
+  }
+
+  export class Building
```

Comment:
```suggestion
  export class Builder
```

This ensures alignment with the builder pattern (https://en.wikipedia.org/wiki/Builder_pattern), which we also employ in other packages.